src/steam_network/user_info_cache.py

```diff
@@ -1,58 +1,68 @@
 import asyncio
 import base64
-import logging as log
+import logging
+from typing import Optional
 
+logger = logging.getLogger(__name__)
 
 class UserInfoCache:
     def __init__(self):
-        self._steam_id = None
-        self._account_id = None
-        self._account_username = None
-        self._persona_name = None
-        self._token = None
-        self._two_step = None
-        self._sentry = b''
+        self._steam_id: Optional[int] = None #unique id Steam assigns to the user
+        self._account_username: Optional[str] = None #user name for the steam account.
+        self._persona_name: Optional[str] = None #friendly name the user goes by in their display. It's what we use when saying "logged in" in the integration page.
+        #Note: The tokens below are strings, but they are formatted as JSON Web Tokens (JWT). We can parse them to determine when the refresh token will expire.
+        self._refresh_token : Optional[str] = None #persistent token. Used to log in, despite the fact that we should use an access token. weird quirk in how steam does things.
+        self._access_token : Optional[str] = None #session login token. Largely useless. May be useful in future if steam fixes their login to use an access token instead of refresh token. 
+
+        #self._guard_data : Optional[str] = None #steam guard data. It might no longer be necessary, but i'll save it just in case. Causes issues with mobile confirm, so it's now excluded
+        
         self._changed = False
-        self.old_flow = False
+        
         self.initialized = asyncio.Event()
 
     def _check_initialized(self):
-        if self._steam_id and self._account_id and self._account_username and self._persona_name and self._token:
-            log.info("User info cache initialized")
+        if self.is_initialized():
+            logger.info("User info cache initialized")
             self.initialized.set()
             self._changed = True
 
+    def is_initialized(self) -> bool:
+        #THIS CURRENTLY ENABLES OR DISABLES LOGGING IN FROM CACHE. 
+
+        #type hinting didn't want to place nice if i didn't do it this way. if you can python better than me and get this to properly bool type hint, go for it -BaumherA
+        #return True if (self._steam_id and self._account_username and self._persona_name and self._refresh_token and self._guard_data) else False
+        return True if (self._steam_id and self._account_username and self._persona_name and self._refresh_token) else False
```

Comment:
its ok. You can also use `return bool(self._steam_id and ...)` or `any([self._steam_id, self._account_username, ...)`

Reply:
yeah, that's terrible. I couldn't coerce Python into returning a bool so i just did that, which was explicit, but ugly. i think that's actually `all`, but i get your point. I didn't think to use the boolean constructor, either (i'm used to casts in c/c++ or C# or Java).

Reply:
```suggestion
        return all((self._steam_id, self._account_username, self._persona_name, self._refresh_token))
```

i'm not sure if you've already done this or not but this produces the same result

Reply:
i updated this in the latest commit. 

I've always errored on the side of caution, but maybe you all can clear this up for me. is `None` truthy? whenever it can be null or falsy, i've always done a null check first just to be safe. 
Lmk if this is overkill.
https://github.com/GOG-Nebula/galaxy-integration-steam/pull/1/files#diff-7b915a3c46ed3f455ffce7147cf1f5f1538ca5f9e6d3a2502d2bb32bccad9120R30

Reply:
`None` is not truthy (`all((None, True))` -> `False`)

Reply:
I currently have `return all([self._steam_id is not None, self._account_username is not None, self.account_username, self._persona_name is not None, self._persona_name, self._refresh_token is not None, self._refresh_token])` 

It's likely overkill. lmk. If you make a suggestion i can just commit it here and be fine with it.

Reply:
it def is but if it works best not to mess with it until after

Reply:
you can always use python REPL to quickly check something (just type python or install ipython). Much easier than in compiled languages.

I would do:

```python
required_fields = [self._steam_id, self._account_username, self._persona_name, self._refresh_token]
return all(x is not None for x in required_fields)
```

Reply:
Updated to make it more pythonic but it's still overkill. I'm leaving it for now because i don't think it's worth a commit but it can get cleaned up during the rewrite.

Reply:
ran tests on a more streamlined version using python interactive, it works as expected. That's been pushed.